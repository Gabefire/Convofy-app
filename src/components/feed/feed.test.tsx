import Feed from "./feed";
import { screen, render } from "@testing-library/react";
import "@testing-library/jest-dom";
import postType from "../../types/post";

describe("feed component", () => {
  let messages: postType[];
  beforeEach(() => {
    const message1: postType = {
      uid: "123",
      from: "gabe",
      content: "test",
      date: new Date(),
      title: "test",
      upVotes: [],
      downVotes: [],
      id: "1",
      forum: "gabe",
      iconURL: null,
    };
    messages = [message1];
  });

  it("renders feed component", async () => {
    const { container } = render(<Feed posts={messages} home={true} />);
    expect(container).toMatchSnapshot();
  });

  it("up vote button on screen", () => {
    render(<Feed posts={messages} home={true} />);
    const upVote = screen.getByRole("button", { name: "up vote" });
    expect(upVote).toBeInTheDocument();
  });
});
