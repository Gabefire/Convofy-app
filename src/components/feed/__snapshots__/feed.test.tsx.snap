// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`feed component renders feed component 1`] = `
<div>
  <div
    id="messages"
  >
    <div
      class="message"
      id="message-1"
    >
      <div
        class="message-title"
      >
        <div
          class="forum-message"
        >
          <div
            class="icon-message"
            style="background-color: red;"
          >
            g
          </div>
          <div
            class="forum-name"
            data-testid="gabe"
          >
            r/gabe
          </div>
        </div>
        <div
          class="from"
        >
          Posted by u/gabe Now
        </div>
      </div>
      <div
        class="message-content"
      >
        <h4>
          test
        </h4>
        <div>
          test
        </div>
      </div>
      <div
        class="bottom-icons"
      >
        <div
          class="likes"
          data-testid="likes"
          id="likes-1"
        >
          <button
            aria-label="up vote"
            class="up-vote-btn arrow-btn"
            id="up-vote-btn-1"
          >
            <img
              class="arrow"
              src="test-file-stub"
            />
          </button>
          0
          <button
            aria-label="down vote"
            class="down-vote-btn arrow-btn"
            id="down-vote-btn-1"
          >
            <img
              class="arrow"
              src="test-file-stub"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
